INSVCTwoFluidStaggeredHierarchyIntegrator.cpp

```diff
@@ -495,9 +496,9 @@ INSVCTwoFluidStaggeredHierarchyIntegrator::preprocessIntegrateHierarchy(const do
 
     // Set the forcing data if applicable.
     d_rhs_vec->setToScalar(0.0);
-    if (d_f_un_fcn) d_f_un_fcn->setDataOnPatchHierarchy(f_un_idx, d_f_un_sc_var, d_hierarchy, current_time);
-    if (d_f_us_fcn) d_f_us_fcn->setDataOnPatchHierarchy(f_us_idx, d_f_us_sc_var, d_hierarchy, current_time);
-    if (d_f_p_fcn) d_f_p_fcn->setDataOnPatchHierarchy(f_p_idx, d_f_cc_var, d_hierarchy, current_time);
+    if (d_f_un_fcn) d_f_un_fcn->setDataOnPatchHierarchy(f_un_idx, d_f_un_sc_var, d_hierarchy, half_time);
+    if (d_f_us_fcn) d_f_us_fcn->setDataOnPatchHierarchy(f_us_idx, d_f_us_sc_var, d_hierarchy, half_time);
+    if (d_f_p_fcn) d_f_p_fcn->setDataOnPatchHierarchy(f_p_idx, d_f_cc_var, d_hierarchy, half_time);
 
```

Comment:
Note to self: we're using half_time because trapezoidal rule has 0.5*( f(t_n)+f(t_n+1) ) on the RHS.

Reply:
We're using the trapezoidal rule for diffusion, here we're using the midpoint rule for forcing, $u_{n+1} = u_n + dt f(t_{n+1/2})$